aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp

```diff
@@ -0,0 +1,250 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDADataType.h>
+#include <ATen/cuda/CUDASparse.h>
+#include <ATen/cuda/CUDAConfig.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/Dispatch.h>
+#include <c10/core/ScalarType.h>
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/util/CallOnce.h>
+#include <c10/util/Half.h>
+#include <cusparse.h>
+#include <cstdint>
+#include <Exceptions.h>
+
+
+#if AT_CUSPARSELT_ENABLED()
+#include <cusparseLt.h>
+#endif
+
+namespace at {
+namespace native {
+
+#if AT_CUSPARSELT_ENABLED()
+cusparseLtHandle_t handle;
+#endif
+
+at::Tensor _cslt_compress(const Tensor& sparse_input)
+{
+#if AT_CUSPARSELT_ENABLED()
+
+    TORCH_CUDASPARSE_CHECK(cusparseLtInit(&handle));
```

Comment:
Shouldn't this happen only once during CUDA initialization?

Reply:
Yeah, I can throw this in a _init_cslt() function and call that if we've compiled with cuSPARSELt. I tried adding this properly using DeviceThreadHandles like cusparse [here](https://github.com/pytorch/pytorch/blob/78fffe890627bc020c9bd82a27a19d042f31695e/aten/src/ATen/cuda/CuSparseHandlePool.cpp#L8). But then I was running into compilation errors because C++ could not convert the handle type to a bool, which causes an error [here](https://github.com/pytorch/pytorch/blob/78fffe890627bc020c9bd82a27a19d042f31695e/aten/src/ATen/cuda/detail/DeviceThreadHandles.h#L52).

Reply:
Could not return the returned value from cusparseLtInit you mean?
If it is a pointer, then that check should properly check that it is not null.